pkg/app/app.go

```diff
@@ -65,41 +65,40 @@ func ValidateFlags(bundle string, args []string, cmd *cobra.Command, opt *kobjec
 	logrus.Debug("Checking validation of provider %s", provider)
 
 	// OpenShift specific flags
-	deploymentConfig := cmd.Flags().Lookup("deployment-config").Changed
 	buildRepo := cmd.Flags().Lookup("build-repo").Changed
 	buildBranch := cmd.Flags().Lookup("build-branch").Changed
 
 	// Kubernetes specific flags
 	chart := cmd.Flags().Lookup("chart").Changed
-	daemonSet := cmd.Flags().Lookup("daemon-set").Changed
-	replicationController := cmd.Flags().Lookup("replication-controller").Changed
-	deployment := cmd.Flags().Lookup("deployment").Changed
 
 	// Check validations against provider flags
 	switch {
 	case provider == "openshift":
 		if chart {
 			logrus.Fatalf("--chart, -c is a Kubernetes only flag")
 		}
-		if daemonSet {
-			logrus.Fatalf("--daemon-set is a Kubernetes only flag")
+		if opt.CreateDS {
+			logrus.Fatalf("DaemonSet is a Kubernetes only controller")
 		}
-		if replicationController {
-			logrus.Fatalf("--replication-controller is a Kubernetes only flag")
+		if opt.CreateRC {
+			logrus.Fatalf("ReplicationController is a Kubernetes only controller")
 		}
-		if deployment {
-			logrus.Fatalf("--deployment, -d is a Kubernetes only flag")
+		if opt.CreateD {
```

Comment:
we should not be looking at `opt`. 

Code such as
```
cmd.Flags().Lookup("replication-controller").Changed
```

Was used to see if the user has _changed_ the value or not. By using `opt` we would get a false reading.